koku/masu/external/downloader/azure/azure_report_downloader.py

```diff
@@ -409,6 +412,65 @@ def manifest_date_format(self):
         """Set the Azure manifest date format."""
         return "%Y%m%d"
 
+    def _check_size(self, key, check_inflate=False):
+        """Check the size of an Azure blob file.
+
+        Determine if there is enough local space to download and decompress the
+        file.
+
+        Args:
+            key (str): the key name of the blob to check
+            check_inflate (bool): if the file is compressed, evaluate the file's decompressed size.
+
+        Returns:
+            (bool): whether the file can be safely stored (and decompressed)
+        """
+        size_ok = False
+
+        try:
+            blob = self._azure_client.get_file_for_key(key, self.container_name)
+            file_size = blob.size
+        except ClientError as ex:
+            if ex.response["Error"]["Code"] == "AccessDenied":
+                msg = f"Unable to access Azure container {self.container_name}: (AccessDenied)"
+                LOG.info(log_json(self.tracing_id, msg=msg, context=self.context))
+                raise AzureReportDownloaderNoFileError(msg)
+            msg = f"Error downloading file: Error: {str(ex)}"
+            LOG.error(log_json(self.tracing_id, msg=msg, context=self.context))
+            raise AzureReportDownloaderError(str(ex))
+
+        if file_size < 0:
+            raise AzureReportDownloaderError(f"Invalid size for Azure blob: {key}")
+
+        total, used, free = shutil.disk_usage(self._get_exports_data_directory())
+        if file_size < free:
+            size_ok = True
+
+        LOG.debug("%s is %s bytes; Download path has %s free", key, file_size, free)
+
+        ext = os.path.splitext(key)[1]
+        if ext == ".gz" and check_inflate and size_ok and file_size > 0:
+            try:
+                range_start = file_size - 4
+                range_end = file_size
+
+                resp = self._azure_client.download_file(
```

Comment:
Having a uncompressed check similar to AWS is ok I'm not sure its required though, since its actually pandas that handles the file in memory. It eventually writes to a csv file but during daily archiving, so we reduce the file down to required columns and by date.
That said if we do want to keep this I would rework it so we don't download the file twice, in and outside of this method. Lets download the file, then pass that file to the compression check logic.